### PR TITLE
Implement edit image annotation shape + video selection

### PR DESCRIPTION
**Pull request by nikolas**
This includes a big refactoring: implement SelectionForm, which the
creation from and edit form both inherit.

### CODE REVIEW

File: media/js/src/assetDetail/AssetDetail.jsx [hunk 1]
```diff
@@ -525,7 +566,7 @@ export default class AssetDetail extends React.Component {
                                     <div className="col-md-9">
                                         <div className="input-group">
 
-                                            {this.state.tab === 'createSelection' && (
+                                            {(this.state.tab === 'createSelection' || this.state.isEditing) && (
```
Comment (sdreher) [inline media/js/src/assetDetail/AssetDetail.jsx:569]: The selection start & selection end buttons & timecode spinners show correctly in edit mode. But, the textual range display is also showing up. Looking at line 585, maybe a `!isEditing` needs to be added?

File: media/js/src/forms/SelectionForm.jsx [hunk 2]
```diff
@@ -0,0 +1,228 @@
...
+                    !hasSelection(this.props.selectionSource)
+            ) || (
+                this.props.type === 'video' &&
+                    !this.props.selectionStartTime &&
+                    !this.props.selectionEndTime
+            )
+        ) {
+            e.preventDefault();
+            e.stopPropagation();
+            this.props.onShowValidationError(noSelectionError);
```
Comment (sdreher) [inline media/js/src/forms/SelectionForm.jsx:83]: For video editing, the code is falling into this path for valid selection start & end ranges. I'm not quickly seeing why this is the case.

Comment (nikolas) [reply media/js/src/forms/SelectionForm.jsx:83]: This is resolved - the video selection props weren't getting passed in to this new form.

**Comment by nikolas** [comment]
These problems are fixed now.

**Pull request merged**